earthfile2llb/converter.go

```diff
@@ -271,6 +271,10 @@ func (c *Converter) FromDockerfile(ctx context.Context, contextPath string, dfPa
 
 // Locally applies the earthly Locally command.
 func (c *Converter) Locally(ctx context.Context, platform *specs.Platform) error {
+	if !c.opt.AllowLocally {
+		return errors.New("LOCALLY cannot be used when --strict is specified")
+	}
+
```

Comment:
This is in converter, because it (to me) helps interpreter stay clean. Interpreter is for language, Converter is where CLI options + interpreted language options are combined, and so also is where the relevant error lives.